src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs

```diff
@@ -243,14 +313,11 @@ protected override void ProcessRecord()
             // use passed date object if specified
             if (_dateSpecified)
             {
-                if (FromUnixTime.IsPresent)
-                {
-                    dateToUse = DateTimeOffset.FromUnixTimeSeconds(Date.Ticks).UtcDateTime;
-                }
-                else
-                {
-                    dateToUse = Date;
-                }
+                dateToUse = Date;
+            }
+            else if (_unixTimeSecondsSpecified)
+            {
+                dateToUse = DateTimeOffset.FromUnixTimeSeconds(UnixTimeSeconds).LocalDateTime;
```

Comment:
Do you consider `-AsUTC ` too?

Reply:
As an aside, @iSazonov: Do you know why `-AsUTC` cannot be combined with `-UFormat`? Was that intentional?

Reply:
I think it can be in UFormat set too.

Reply:
The AsUTC parameter can be combined with all other parameters.

Reply:
@aetos382 Please add AsUTC support.

Reply:
It's supported here (it's the same code path as using -Date -AsUTC).

```cs
if (AsUTC)
{
    dateToUse = dateToUse.ToUniversalTime();
}
```

Reply:
I mean why do you interpret `UnixTimeSeconds` as local Datetime here?

Reply:
Because `dateToUse` is local time (it's not `DateTime.UtcNow`).

```cs
protected override void ProcessRecord()
{
    DateTime dateToUse = DateTime.Now;
```

This command consistently handles the local time unless an `-AsUTC` parameter is set.

If the `-AsUTC` parameter is set, it will be handled by the code cited in the above comment, and therefore it has not been considered here.

Reply:
I agree that outputting a local date by default makes sense with `-FromUnixTime` too, but to provide some background:

`Get-Date` _defaults to_ a local `[datetime]` instance (one whose `.Kind` property value is `Local`),  (`(Get-Date).Kind`) and you also get such an instance if you modify the current date with parameters such as `-Year`
 (e.g., `(Get-Date -Year 2019).Kind`).

By contrast, if you pass an argument to `-Date`:

* If you pass a _preexisting_ `[datetime]` instance, its `.Kind` property is retained (e.g.
`(Get-Date ([datetime]::UtcNow)).Kind` (`Utc`)

* If you pass a _string_ that is implicitly _converted to_ `[datetime]` _during parameter binding_ (that is, this logic is outside of `Get-Date`):
   * If that string _doesn't_ contain a time-zone offset, the `.Kind` property is `Unspecified`; e.g.,
 `(Get-Date 2020-01-01).Kind` (`Unspecified`) 
    * If it _does_ contain a time-zone offset (whatever it is, even if it indicates UTC), the `.Kind` property is `Local`; that is, _conversion to local time_ takes place; e.g., 
`(Get-Date 2020-01-01Z).Kind` (converted from UTC to `Local`).

With respect to string representations and `.ToUniversalTime()`, `Unspecified` instances behave like `Local` ones.
However, calling `.ToLocalTime()` on an `Unspecified` instance interprets it as _UTC_ time.

Reply:
Ok, if we accept the default we should add a test and document this.

Reply:
Hmm...

Would it be better to have `-Kind <DateTimeKind>` parameter instead of an `-AsUTC`?

Reply:
I guess the DateTimeKind was added to mitigate Datetime type limitations (today we could use DateTimeOffset). I do not think that we should use DateTimeKind broadly. At least we could discuss this in new issue.

Reply:
In fact, there already _is_ a proposal to implement this: #11731 - and I do think it makes perfect sense.

Whether or not we implement it as part of this PR, I think it's important to also include it in v7.1

Reply:
I think a parameter like `-AsTimeZone PDT` would be useful.

Then I would also want features like `-FromTimeZone PDT -ToTimeZone JST`.

But adding those features to `Get-Date` would make the command overly complicated.
For example, I think we should add another command, such as `Convert-Time`.

Should there be a place for a comprehensive discussion on how `Get-Date` should handle timezone and offset-related parameters (including `-AsUTC`)?

Reply:
Let's continue in  #11731.

Reply:
@iSazonov: See my comment at https://github.com/PowerShell/PowerShell/issues/11731#issuecomment-659279816.

I think that potential future `[datetimeoffset]` and time-zone (conversion) support should be discussed in a _new_ issue, which I encourage you, @aetos382, to create (as for the name `Convert-Time`: for consistency it's better to stick with the word "date").

Reply:
@aetos382 You can open new issue to discuss Convert-Time/Date/Datetime cmdlet.